dask/compose.py

```diff
@@ -0,0 +1,281 @@
+import operator
+from functools import partial, wraps
+from itertools import chain
+from collections import Iterator
+
+from toolz import merge, unique
+
+from .core import istask
+from .optimize import cull, fuse
+from .context import _globals
+from .compatibility import apply
+from . import threaded
+
+__all__ = ['do', 'value', 'Value']
+
+
+def flat_unique(ls):
+    """Flatten ``ls``, filter by unique id, and return a list"""
+    return list(unique(chain.from_iterable(ls), key=id))
+
+
+def unzip(ls, nout):
+    """Unzip a list of lists into ``nout`` outputs."""
+    out = list(zip(*ls))
+    if not out:
+        out = [()] * nout
+    return out
+
+
+def to_task_dasks(task):
+    """Normalize a task and extract all sub-dasks.
+
+    - Replace ``Values`` with their keys
+    - Convert literals to things the schedulers can handle
+    - Extract dasks from all enclosed values
+
```

Comment:
This function could use some doctests to show how it works.  I think that it becomes very clear once you see

```
>>> to_task_dasks((1, 2, 3))
(tuple, [1, 2, 3])
```